src/Datadog.Trace.ClrProfiler.Managed/Integrations/HttpMessageHandlerIntegration.cs

```diff
@@ -21,54 +22,93 @@ public static class HttpMessageHandlerIntegration
+            var handlerType = typeof(HttpMessageHandler); // Note the HttpMessageHandler to match the method call we replaced
+
+            var sendAsync = Emit.DynamicMethodBuilder<Func<HttpMessageHandler, HttpRequestMessage, CancellationToken, Task<HttpResponseMessage>>>
+                                .GetOrCreateMethodCallDelegate(
+                                     handlerType,
+                                     "SendAsync",
+                                     callOpCode);
+
+            return SendAsyncInternal(
+                sendAsync,
+                callOpCode == OpCodeValue.Call ? handlerType : handler.GetType(),
+                (HttpMessageHandler)handler,
+                (HttpRequestMessage)request,
+                cancellationToken);
+        }
+
+        /// <summary>
+        /// Instrumentation wrapper for <see cref="HttpMessageHandler.SendAsync"/>.
+        /// </summary>
+        /// <param name="handler">The <see cref="HttpMessageHandler"/> instance to instrument.</param>
+        /// <param name="request">The <see cref="HttpRequestMessage"/> that represents the current HTTP request.</param>
+        /// <param name="cancellationTokenSource">The <see cref="CancellationTokenSource"/> that can be used to cancel this <c>async</c> operation.</param>
+        /// <param name="opCode">The OpCode used in the original method call.</param>
+        /// <returns>Returns the value returned by the inner method call.</returns>
         [InterceptMethod(
             TargetAssembly = "System.Net.Http",
             TargetType = "System.Net.Http.HttpClientHandler",
+            TargetMethod = "SendAsync",
             TargetMinimumVersion = Major4,
-            TargetMaximumVersion = Major4)] // .NET Framework and .NET Core 2.0 and earlier
-            /*
-        [InterceptMethod(
-            TargetAssembly = "System.Net.Http",
-            TargetType = "System.Net.Http.SocketsHttpHandler")] // .NET Core 2.1 and later
-        */
-        public static object SendAsync(
+            TargetMaximumVersion = Major4)]
+        public static object HttpClientHandler_SendAsync(
             object handler,
             object request,
-            object cancellationTokenSource)
+            object cancellationTokenSource,
+            int opCode)
         {
-            // HttpMessageHandler
-            // Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+            // original signature:
+            // Task<HttpResponseMessage> HttpClientHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
             var tokenSource = cancellationTokenSource as CancellationTokenSource;
             var cancellationToken = tokenSource?.Token ?? CancellationToken.None;
+            var callOpCode = (OpCodeValue)opCode;
+            var handlerType = typeof(HttpClientHandler); // Note the HttpClientHandler to match the method call we replaced
```

Comment:
Could you rename `handlerType` to `httpClientHandlerType` to be a little more specific about the meaning of the variable? I was confused initially because I thought `handlerType` meant `handler.GetType()`

Reply:
Renamed that and a few more in feffd93784f6c9937e21f052628258a4affe766e.

Reply:
That helps a lot, thanks!